2022Swerve/src/main/java/frc/robot/subsystems/AutonSubsytem.java

```diff
@@ -0,0 +1,137 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryUtil;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.StringLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.SwerveControllerCommand;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.Constants.AutonConstants;
+import frc.robot.Constants.DriveConstants;
+
+
+public class AutonSubsytem extends SubsystemBase{
+    private enum AutonModes{
+        FORWARD, // Go forward 2 meters
+        BACKWARD, // Wait 3 seconds, go backward 2 meters
+        FORWARD180; // Go forward 2 meters and rotate 180 degrees
+    }
+
+    private ShuffleboardTab configTab = Shuffleboard.getTab("Config");
```

Comment:
Where did we have the chooser last year?  Was it on a separate tab than the competition console?

Let's put this where we think it will go in the end.  Want to try and reduce the number of last minute touch ups we have to do this year.

Reply:
The chooser was in Robot.java, although I personally dislike this approach. This is because AutonSubsystem was directly imported to Robot.java, and there was a significant amount of code in Robot and RobotContainer that would probably be better contained in the subsystem.

Reply:
Sorry, didn't mean the code location, I meant the tab in shuffleboard.  Was it "Config" ?

Reply:
It was in config, yes

Reply:
k, resolve